lib/ansible/modules/database/mysql/mysql_variables.py

```diff
@@ -186,10 +244,30 @@ def main():
         # Type values before using them
         value_wanted = typedvalue(value)
         value_actual = typedvalue(mysqlvar_val)
+        value_in_auto_cnf = None
+        if var_in_mysqld_auto_cnf is not None:
+            value_in_auto_cnf = typedvalue(var_in_mysqld_auto_cnf)
+
         if value_wanted == value_actual:
-            module.exit_json(msg="Variable already set to requested value", changed=False)
+            if mode == 'persist':
+                if value_in_auto_cnf is None:
+                    module.warn("The current global variable '%s' has value '%s' "
```

Comment:
I wouldn't emit this warning. After all, the warning is just saying "the module is doing its job, nothing out of the ordinary".